tools/api.py

```diff
@@ -1057,9 +1064,12 @@ def add_cpp_class(self) -> None:
     ${output_decls}
     if (${fname} == nullptr)
         throw std::runtime_error("${name} function is missing.");
+    std::array<char, 256> exception_msg;
```

Comment:
This should initialize `exception_msg.front()` to null so if a message is not written there will be valid empty string.

Reply:
Done.